### ISSUE TITLE
hakyll build fails with "cache error" on "templates/default.html"

### ISSUE THREAD

**Issue by emlautarom1** [jaspervdj/hakyll#766]
System specs:
- **ghc**: 8.8.3
- **cabal**: 3.0.0.0
- **OS**: Windows 10 v1909

Steps to reproduce:

```
mkdir HakyllSite
hakyll-init -f HakyllSite
cd HakyllSite
cabal run exe:site -- build
```

The output I get is the following:

```
Initialising...
  Creating store...
  Creating provider...
  Running rules...
Checking for out-of-date items
Compiling
  [ERROR] Hakyll.Core.Compiler.Require.load: templates\default.html (snapshot 
_final) was not found in the cache, the cache might be corrupted or the item you are referring to might not exist
```

I can confirm that the `templates\default.html` file exists, so the issue must be somewhere else.

**Comment by emlautarom1** [comment]
**Note:** Deleting the file `templates\default.html` makes no difference to the error, so I think the root of the problem could be on the early `fail notFound`:

https://github.com/jaspervdj/hakyll/blob/b08dd317a381a0436607b44257c487b1930dcc72/lib/Hakyll/Core/Compiler/Require.hs#L55-L63

**Comment by LaurentRDC** [comment]
I'm also encountering this problem right now, on every template file.

**Comment by LaurentRDC** [comment]
I'm getting the same type of error for every template. I am also on Windows.

This error did not occur with Hakyll 4.12.4.0 (stackage lts-12.26). The error occurs when I update to Hakyll 4.13.3.0 (stackage nighlty-05-22-2020).

Looking at the changelog, there is this:

> # Hakyll 4.13.1.0 (2020-02-26)
(...)
> * Fix file path compatibility with Windows (by Hexirp)
(...)

I wonder if that's the source of the error?

**Comment by LaurentRDC** [comment]
Ok I've looked into it a little, and I found that all files that are not in the current working directory are ignored.

For the `hakyll-init` site, this includes files in `posts/`, `templates/`, `css/`, etc.

I'll keep looking.

**Comment by jaspervdj** [comment]
@LaurentRDC thanks for looking into this!  Let me know if there’s anything I can do.

**Comment by LaurentRDC** [comment]
So I have found the source of the problem. Pattern matching (via the functions `Hakyll.Core.Identifier.Pattern.matches` and `Hakyll.Core.Identifier.Pattern.capture` does not work on Windows. This is probably due to the difference between path separators on Windows vs. Posix.

I'll try to make it work and create a pull request soon. My approach is that paths (globs or literal) should be normalized by using `System.FilePath.normalize`.

**Comment by jaspervdj** [comment]
Fixed in #778
